program.go

```diff
@@ -55,3 +55,83 @@ func (p *Program) Streams() (ss []*Stream) {
 	}
 	return
 }
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a8c87564167b87f54be9171778d51fe49
+func (p *Program) Flags() int {
+	return int(p.c.flags)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a8c87564167b87f54be9171778d51fe49
+func (p *Program) SetFlags(f int) {
+	p.c.flags = C.int(f)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a9c7a07c08a1f960aaa49f3f47633af5c
+func (p *Program) Discard() Discard {
+	return Discard(p.c.discard)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a9c7a07c08a1f960aaa49f3f47633af5c
+func (p *Program) SetDiscard(d Discard) {
+	p.c.discard = int32(d)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a7967d41af4812ed61a28762e988c7a02
+func (p *Program) StreamIndex() uint {
+	return uint(C.uint(*p.c.stream_index))
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#ae9dab38d4694e3da9cba0f882f4e43d3
+func (p *Program) Metadata() *Dictionary {
+	return newDictionaryFromC(p.c.metadata)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#ae9dab38d4694e3da9cba0f882f4e43d3
+func (p *Program) SetMetadata(d *Dictionary) {
+	p.c.metadata = d.c
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a4c1539ea3c98da979b95a59a3ea163cb
+func (p *Program) ProgramNum() int {
+	return int(p.c.program_num)
+}
+
+// https://ffmpeg.org/doxygen/7.0/structAVProgram.html#a4c1539ea3c98da979b95a59a3ea163cb
+func (p *Program) SetProgramNum(n int) {
+	p.c.program_num = C.int(n)
+}
+
```

Comment:
Could you add setters for all the methods below and use them to test all your new methods?

Reply:
Can you make sure to take care of this comment?

Reply:
i think all new methods has been tested.

Reply:
> Can you make sure to take care of this comment?

i think all new methods has been tested.

Reply:
I'd rather have setters for all getters you've added (`SetStreamIndex()`, `SetPmtPid()`, etc.) and use them in tests:
```go
p.SetPmtPid(1)
require.Equal(t, 1, p.PmtPid())
```

Reply:
I deleted `PmtVersion()` method.
 `start_time end_time pts_wrap_reference pts_wrap_behavior` fields can not be changed, so they may not have setters.

Reply:
OK 👍  I think the only missing setter is `SetStreamIndex()` and then we'll be good to merge the PR 👍

Reply:
`program.stream_index` and `stream.index` must be consistent.